shared/src/hooks/useFetchLiquidityGaugeV5Data.ts

```diff
@@ -0,0 +1,181 @@
+import { isProduction } from "../utils/env";
+import {
+  RibbonTokenAddress,
+  VaultLiquidityMiningMap,
+  VaultOptions,
+} from "../constants/constants";
+import { BigNumber } from "@ethersproject/bignumber";
+import { getLiquidityGaugeV5 } from "./useLiquidityGaugeV5";
+import { getERC20Token } from "./useERC20Token";
+import { getERC20TokenNameFromVault } from "../models/eth";
+import useLiquidityTokenMinter from "./useLiquidityTokenMinter";
+import useLiquidityGaugeController from "./useLiquidityGaugeController";
+
+const useFetchLiquidityGaugeV5Data = (): LiquidityGaugeV5PoolData => {
+  const { active, chainId, account: web3Account, library } = useWeb3React();
+  const { provider } = useWeb3Context();
+  const account = impersonateAddress ? impersonateAddress : web3Account;
+  const { transactionsCounter } = usePendingTransactions();
+  const minterContract = useLiquidityTokenMinter();
+  const gaugeControllerContract = useLiquidityGaugeController();
+
+  const [data, setData] = useState<LiquidityGaugeV5PoolData>(
+    defaultLiquidityGaugeV5PoolData
+  );
+  const [, setMulticallCounter] = useState(0);
+
+  const doMulticall = useCallback(async () => {
+    if (!chainId || !minterContract || !gaugeControllerContract) {
+      return;
+    }
+
+    if (!isProduction()) {
+      console.time("Liquidity Gauge V5 Data Fetch");
+    }
+
+    /**
+     * We keep track with counter so to make sure we always only update with the latest info
+     */
+    let currentCounter: number;
+    setMulticallCounter((counter) => {
+      currentCounter = counter + 1;
+      return currentCounter;
+    });
+
+    const minterResponsePromises = Promise.all([
+      minterContract.rate(),
+      gaugeControllerContract.time_total(),
+    ]);
+
+    const guageResponsesPromises = Promise.all(
+      Object.keys(VaultLiquidityMiningMap.lg5).map(async (_vault) => {
+        const vault = _vault as VaultOptions;
+
+        const lg5Contract = getLiquidityGaugeV5(
+          library || provider,
+          vault,
+          active
+        )!;
+        const vaultContract = getERC20Token(
+          library || provider,
+          getERC20TokenNameFromVault(vault, "v2"),
+          chainId,
+          active
+        );
+
+        /**
+         * 1. Pool size
+         * 2. Relative weight
+         */
+        const unconnectedPromises: Promise<BigNumber>[] = [
+          lg5Contract.totalSupply(),
+          gaugeControllerContract["gauge_relative_weight(address)"](
+            VaultLiquidityMiningMap.lg5[vault]!
+          ),
+        ];
+
+        /**
+         * 1. Current stake
+         * 2. Claimable rbn
+         * 3. Unstaked Balance
+         * 4. Claimed rbn
+         */
+        const promises = unconnectedPromises.concat(
+          active
+            ? [
+                lg5Contract.balanceOf(account!),
+                lg5Contract.claimable_reward(account!, RibbonTokenAddress),
+                vaultContract!.balanceOf(account!),
```

Comment:
This needs to be `shares` instead of `balanceOf` because `balanceOf` refers to the shares held by the user's account. In v2 shares can also be held on behalf of the user by the vault.